Bump versions of maintenance-packages dependencies consumed in runtime

carlossanlop opened the pull request:
dotnet/runtime depends on these packages that we are now publishing from dotnet/maintenance-packages:

- Microsoft.Bcl.HashCode
- System.Buffers
- System.Memory
- System.Threading.Tasks.Extensions
- System.Runtime.CompilerServices.Unsafe

Bumping their versions to consume the new preview versions, available in the dotnet-libraries feed: https://dnceng.visualstudio.com/public/_artifacts/feed/dotnet-libraries

Related PRs:
- machinelearning: https://github.com/dotnet/runtime/pull/108806
- winforms: https://github.com/dotnet/winforms/pull/12313

carlossanlop committed:
Bump dependency versions of packages that now ship out of maintenance-packages (preview)

- Microsoft.Bcl.HashCode
- System.Buffers
- System.Memory
- System.Threading.Tasks.Extensions
- System.Runtime.CompilerServices.Unsafe

ericstj commented on eng/Versions.props:
```diff
@@ -129,9 +129,9 @@
+    <SystemRuntimeCompilerServicesUnsafeVersion>6.1.0-preview.1.24511.1</SystemRuntimeCompilerServicesUnsafeVersion>
     <SystemThreadingAccessControlVersion>7.0.0</SystemThreadingAccessControlVersion>
-    <SystemThreadingTasksExtensionsVersion>4.5.4</SystemThreadingTasksExtensionsVersion>
+    <SystemThreadingTasksExtensionsVersion>4.6.0-preview.1.24511.1</SystemThreadingTasksExtensionsVersion>
```
Looks to me like CLSCompliant attribute value on this assembly is causing errors.  @carlossanlop you double-check that the CLSCompliant attribute correct on all the m-p OOBs?

carlossanlop replied:
Yes.

carlossanlop replied:
Just for my own peace of mind, I'll post screenshot comparisons of them all. Give me some time to collect them.

carlossanlop committed:
Bump to latest with CLSCompliant fix.

carlossanlop commented:
There's a CI failure indicating that a triple slash cref pointing to `MathF` is ambiguous. I see that the updated `System.Numerics.Vectors` OOB package defines `MathF` but it is an internal class there. I don't see how it could be conflicting.

carlossanlop commented:
@ericstj @michaelgsharp @ViktorHofer  I don't think the problem is in the package. When comparing the DLLs with ILSpy, before and after maintenance-packages, they all contain the `MathF` class and it's _internal_. So I don't know what could be causing the cref conflict.

The old nupkg:
- net462
![image](https://github.com/user-attachments/assets/d8f83ce7-76e6-4d3b-b2ac-532e421b1ead)
- netstandard2.0
![image](https://github.com/user-attachments/assets/a4a1cd0c-8b2a-4c6a-99ad-b062fc5a5ee8)

The new preview nupkg:
- net462:
![image](https://github.com/user-attachments/assets/0f4a685a-777f-4c0d-8758-83f9a9c9b52e)
- netstandard2.0:
![image](https://github.com/user-attachments/assets/e4ce400f-73a7-419f-8d3f-6c066e555be4)

ericstj commented:
The old package contained a reference assembly, which didn't include the internal `MathF` type.  That's why the new package is hitting this.  The new package only exposes the implementation, which has the internal type.  This is issue https://github.com/dotnet/roslyn/issues/71442.

We can workaround it by aliasing the problematic assembly, and then referring to it by the alias.  I'll submit a fix.

ericstj committed:
Workaround name conflict for MathF

See https://github.com/dotnet/roslyn/issues/71442 We refer to MathF from
a comment which hits this bug problem in Roslyn where it thinks it's
ambiguous because visibility is not considered when resolving cref's in
doc comments.

carlossanlop commented:
Thanks for the fix @ericstj

>  I doubt customers will hit this bug

So there's no way a customer will do what Tensors is doing? Is Tensors special and unique enough because of the way runtime works and consumes the package?

carlossanlop commented:
@MichaelSimons what specific thing abouy these nupkgs is considered a prebuilt?
We are hitting this error:

```
error : 6 new pre-builts discovered! 
Detailed usage report can be found at /__w/1/s/artifacts/sb/prebuilt-report/baseline-comparison.xml. 
See https://aka.ms/dotnet/prebuilts for guidance on what pre-builts are and how to eliminate them. 
Package IDs are: 
Microsoft.Bcl.HashCode.6.0.0-preview.1.24517.1 
System.Buffers.4.6.0-preview.1.24517.1 
System.Memory.4.6.0-preview.1.24517.1 
System.Numerics.Vectors.4.6.0-preview.1.24517.1 
System.Runtime.CompilerServices.Unsafe.6.1.0-preview.1.24517.1 
System.Threading.Tasks.Extensions.4.6.0-preview.1.24517.1
```

ericstj commented:
> So there's no way a customer will do what Tensors is doing?

It's not impossible, just unlikely for a customer to `cref` `MathF` from their doc comments.

> what specific thing about these nupkgs is considered a prebuilt?

They're built from a repo that's not part of source build yet, and they aren't provided by SBRP.  Maybe we should bring maintenance-packages into source build @MichaelSimons.  In the meantime I think we can mark these as permitted prebuilts.

ericstj committed:
Permit maintenance-packages prebuilts

MichaelSimons commented:
> They're built from a repo that's not part of source build yet, and they aren't provided by SBRP. Maybe we should bring maintenance-packages into source build @MichaelSimons. In the meantime I think we can mark these as permitted prebuilts.

If you do this the runtime flow will fail in the sdk repo until the dotnet/maintenance-packages repo is onboarded onto SB.  If you don't want to block the flow I would wait merging this PR until you have onboarded dotnet/maintenance-packages into source-build.

carlossanlop commented on eng/Versions.props:
```diff
@@ -111,16 +111,17 @@
     <!-- Libraries dependencies -->
     <MicrosoftBclAsyncInterfacesVersion>6.0.0</MicrosoftBclAsyncInterfacesVersion>
-    <MicrosoftBclHashCodeVersion>1.1.1</MicrosoftBclHashCodeVersion>
+    <MicrosoftBclHashCodeVersion>6.0.0-preview.1.24517.1</MicrosoftBclHashCodeVersion>
```
```suggestion
    <MicrosoftBclHashCodeVersion>6.0.0-preview.1.24529.4</MicrosoftBclHashCodeVersion>
```

carlossanlop commented on eng/Versions.props:
```diff
@@ -111,16 +111,17 @@
     <MicrosoftWin32RegistryVersion>5.0.0</MicrosoftWin32RegistryVersion>
     <StyleCopAnalyzersVersion>1.2.0-beta.507</StyleCopAnalyzersVersion>
-    <SystemBuffersVersion>4.5.1</SystemBuffersVersion>
+    <SystemBuffersVersion>4.6.0-preview.1.24517.1</SystemBuffersVersion>
```
```suggestion
    <SystemBuffersVersion>4.6.0-preview.1.24529.4</SystemBuffersVersion>
```

carlossanlop commented on eng/Versions.props:
```diff
@@ -111,16 +111,17 @@
     <SystemFormatsAsn1Version>8.0.1</SystemFormatsAsn1Version>
     <SystemIOFileSystemAccessControlVersion>5.0.0</SystemIOFileSystemAccessControlVersion>
-    <SystemMemoryVersion>4.5.5</SystemMemoryVersion>
+    <SystemMemoryVersion>4.6.0-preview.1.24517.1</SystemMemoryVersion>
```
```suggestion
    <SystemMemoryVersion>4.6.0-preview.1.24529.4</SystemMemoryVersion>
```

carlossanlop commented on eng/Versions.props:
```diff
@@ -129,9 +130,9 @@
     <SystemSecurityPermissionsVersion>7.0.0</SystemSecurityPermissionsVersion>
     <SystemTextJsonVersion>10.0.0-alpha.1.24504.10</SystemTextJsonVersion>
-    <SystemRuntimeCompilerServicesUnsafeVersion>6.0.0</SystemRuntimeCompilerServicesUnsafeVersion>
+    <SystemRuntimeCompilerServicesUnsafeVersion>6.1.0-preview.1.24517.1</SystemRuntimeCompilerServicesUnsafeVersion>
```
```suggestion
    <SystemRuntimeCompilerServicesUnsafeVersion>6.1.0-preview.1.24529.4</SystemRuntimeCompilerServicesUnsafeVersion>
```

carlossanlop commented on eng/Versions.props:
```diff
@@ -129,9 +130,9 @@
+    <SystemRuntimeCompilerServicesUnsafeVersion>6.1.0-preview.1.24517.1</SystemRuntimeCompilerServicesUnsafeVersion>
     <SystemThreadingAccessControlVersion>7.0.0</SystemThreadingAccessControlVersion>
-    <SystemThreadingTasksExtensionsVersion>4.5.4</SystemThreadingTasksExtensionsVersion>
+    <SystemThreadingTasksExtensionsVersion>4.6.0-preview.1.24517.1</SystemThreadingTasksExtensionsVersion>
```
```suggestion
    <SystemThreadingTasksExtensionsVersion>4.6.0-preview.1.24529.4</SystemThreadingTasksExtensionsVersion>
```

carlossanlop reviewed (commented):
Bump the versions to the preview used in SBRP.

carlossanlop committed:
Bump versions to preview used in SBRP

carlossanlop commented:
I'll address the conflict after I merge https://github.com/dotnet/runtime/pull/109390 which unblocks this PR (so that I don't restart the CI unnecessarily).

carlossanlop committed:
Add missing SystemNumericsVectorsVersion, it is only used in the workaround for Tensors

carlossanlop committed:
Revert "Permit maintenance-packages prebuilts" as it is not needed anymore after updating SBRP.

This reverts commit fe556303cf5800c2d66437ad53cdc1c6f55ad3e6.

carlossanlop commented:
I triggered a new update from SBRP to bump the Numerics.Vectors package version to the correct one: https://github.com/dotnet/runtime/pull/109561

carlossanlop committed:
Bump to latest prerelease

carlossanlop commented:
@ViktorHofer I updated this PR to also consume the new System.Data.SqlClient prerelease version and there's an ApiCompat failure that makes sense: The old package used to contain a netcoreapp2.1 folder, but the new package contains a net6.0 and a net8.0 folder.

Is this failure expected and supressable, or should we add a placeholder for netcoreapp2.1 in System.Data.SqlClient?

```
src/libraries/apicompat/ApiCompat.proj(93,5): error MSB4018: (NETCORE_ENGINEERING_TELEMETRY=Build) The "Microsoft.DotNet.ApiCompat.Task.ValidateAssembliesTask" task failed unexpectedly.
System.IO.DirectoryNotFoundException: Could not find a part of the path '/__w/1/s/.packages/system.data.sqlclient/4.9.0-rtm.24561.10/lib/netcoreapp2.1/System.Data.SqlClient.dll'.
```

ViktorHofer committed:
Update ApiCompat.proj

carlossanlop committed:
Update to stable versions

carlossanlop commented:
@cheenamalhotra @David-Engel we're updating the SqlClient dependency in runtime to the latest. I suppose the warning is expected since it's now marked as obsolete. Should we suppress it or what do you suggest?

David-Engel commented:
> we're updating the SqlClient dependency in runtime to the latest. I suppose the warning is expected since it's now marked as obsolete. Should we suppress it or what do you suggest?

Suppress, IMO.

cheenamalhotra commented:
> @cheenamalhotra @David-Engel we're updating the SqlClient dependency in runtime to the latest. I suppose the warning is expected since it's now marked as obsolete. Should we suppress it or what do you suggest?

Where is it used? Can you create an issue to track migration to Microsoft.Data.SqlClient in future releases of .NET runtime?

carlossanlop committed:
Add CS0618 to System.Data.Common.Tests.csproj NoWarn list - 'SqlClientFactory' is obsolete: 'Use the Microsoft.Data.SqlClient package instead.

carlossanlop commented on src/libraries/System.Data.Common/tests/System.Data.Common.Tests.csproj:
```diff
@@ -1,6 +1,7 @@
   <PropertyGroup>
-    <NoWarn>$(NoWarn),0168,0169,0414,0219,0649</NoWarn>
+    <!-- 0618 suppresses warning to upgrade to Microsoft.Data.SqlClient. Remove from NoWarn list after upgrading. -->
+    <NoWarn>$(NoWarn),0168,0169,0414,0219,0649,0618</NoWarn>
```
Oh, good point. Alright, changing that.

carlossanlop committed:
Suppress CS0618 on site instead of in csproj

carlossanlop commented:
Sigh. Source build once again complaining about prebuilts:
```
.packages/microsoft.dotnet.arcade.sdk/10.0.0-beta.24556.2/tools/SourceBuild/AfterSourceBuild.proj(81,5): error : (NETCORE_ENGINEERING_TELEMETRY=AfterSourceBuild) 6 new pre-builts discovered! Detailed usage report can be found at /__w/1/s/artifacts/sb/prebuilt-report/baseline-comparison.xml.
See https://aka.ms/dotnet/prebuilts for guidance on what pre-builts are and how to eliminate them.
Package IDs are:
Microsoft.Bcl.HashCode.6.0.0
System.Buffers.4.6.0
System.Memory.4.6.0
System.Numerics.Vectors.4.6.0
System.Runtime.CompilerServices.Unsafe.6.1.0
System.Threading.Tasks.Extensions.4.6.0
```

carlossanlop commented:
The wasm failures are unrelated:
- https://github.com/dotnet/runtime/issues/104349
- A build process died because the machine got reset

MichaelSimons commented:
> Sigh. Source build once again complaining about prebuilts:

It looks like these need to be added to SBRP because of the update to use stable versions in https://github.com/dotnet/runtime/pull/108806/commits/ec8185fa13d240dc36ecbff24d903d638d91368e.  This was the plan discussed in https://github.com/dotnet/maintenance-packages/pull/140#issuecomment-2444383987.

carlossanlop commented:
> It looks like these need to be added to SBRP

Here you go https://github.com/dotnet/source-build-reference-packages/pull/1082

carlossanlop commented:
Darc flow PR https://github.com/dotnet/runtime/pull/110071

akoeplinger committed:
Merge branch 'main' into BumpMaintenancePackagesDependencies
